santorini-common/src/objects/state.rs

```diff
@@ -50,10 +51,23 @@ impl State {
     // }
 
     /// Mutates the game state. Any errors surfaced here are not irrecoverable.
-    fn apply_command(&mut self, command: &dyn Command) -> Result<(), SantoriniError> {
+    pub fn apply_command(&mut self, command: &dyn Command) -> Result<(), SantoriniError> {
         command.execute(self)?;
         Ok(())
     }
+
+    // Returns a winner, assuming that a legal state has exactly one winner
+    pub fn winner(&self) -> Option<Player> {
+        for space in self.board.iter().flatten() {
+            if let Some(worker) = space.worker() {
+                if space.tower().level() == Level::Three {
```

Comment:
dang we can't use && with if let?

Reply:
Doesn't look like it. Also I preferred this expression for clarity.